webhook.go

```diff
@@ -224,17 +239,18 @@ func isSafe(pod *corev1.Pod, operationType string) error {
 }
 
 // Check whether the target resoured need to be mutated
-func mutationRequired(ignoredList []string, metadata *metav1.ObjectMeta) bool {
+func mutationRequired(protectedList []string, metadata *metav1.ObjectMeta) (bool, error) {
 
 	// this output can be used for creating tests/FakeMutationRequiredXXX.json files
-	glog.Infof("mutationRequired log. ignoredList %#v", ignoredList)
+	glog.Infof("mutationRequired log. protectedList %v", protectedList)
 	// logJSON("FakeMutationRequired.json", metadata)
 
-	// skip special kubernete system namespaces
-	for _, namespace := range ignoredList {
-		if metadata.Namespace == namespace {
-			glog.Infof("Skip mutation for %v for it' in special namespace:%v", metadata.Name, metadata.Namespace)
-			return false
+	isProtected := false
+	// skip special kubernetes system namespaces, and protect the TSI namespace
+	for _, protectedNamespace := range protectedList {
```

Comment:
note: okay since protected list is small, if this were to ever become bigger, we should use a map.